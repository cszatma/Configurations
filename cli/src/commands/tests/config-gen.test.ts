import 'jest';

import { runConfigGenCLI } from './test-utils';

const rootPackageJson = require('../../../package.json');

describe('Test general config-gen options', () => {
    it('should print the version number', async () => {
        const { stdout } = await runConfigGenCLI('-v');
        expect(stdout).toBe(rootPackageJson.version + '\n');
    });

    it('should print the help for config-gen', async () => {
        const { stdout } = await runConfigGenCLI('--help');
        expect(stdout).toMatchSnapshot();
    });

    it('should print the help when no arguments are passed', async () => {
        const { stdout } = await runConfigGenCLI('');
        expect(stdout).toMatchSnapshot();
    });

    it('should exit with an error when an unknown command is passed', () => {
        return runConfigGenCLI('does-not-exits').catch(
            ({ code, stderr, stdout }) => {
                expect(code).toBe(1);
                expect(stderr).toBe(
                    'Error: does-not-exits is not a valid command!\n',
                );
                expect(stdout).toBe(
                    'Use `config-gen --help` to see a list of available commands.\n',
                );
            },
        );
    });
});
