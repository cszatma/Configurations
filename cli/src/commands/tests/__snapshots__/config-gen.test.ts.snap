// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test general config-gen options should print the help for config-gen 1`] = `
"
  Usage: config-gen [options] [command]

  Options:

    -v, --version                output the version number
    -h, --help                   output usage information

  Commands:

    add [options] <config-name>  Adds the given configuration file to your project
    list|ls [config]             List all available configs or list available file types for a given config

Run \`config-gen COMMAND --help\` for more information about a given command.
"
`;

exports[`Test general config-gen options should print the help when no arguments are passed 1`] = `
"
  Usage: config-gen [options] [command]

  Options:

    -v, --version                output the version number
    -h, --help                   output usage information

  Commands:

    add [options] <config-name>  Adds the given configuration file to your project
    list|ls [config]             List all available configs or list available file types for a given config

Run \`config-gen COMMAND --help\` for more information about a given command.
"
`;
